distributed/worker.py

```diff
@@ -598,51 +594,16 @@ def __init__(
 
         self._setup_logging(logger)
 
-        if not local_directory:
-            local_directory = (
-                dask.config.get("temporary-directory") or tempfile.gettempdir()
-            )
-        local_directory = os.path.join(local_directory, "dask-worker-space")
-
-        with warn_on_duration(
-            "1s",
-            "Creating scratch directories is taking a surprisingly long time. ({duration:.2f}s) "
-            "This is often due to running workers on a network file system. "
-            "Consider specifying a local-directory to point workers to write "
-            "scratch data to a local disk.",
-        ):
-            self._workspace = WorkSpace(local_directory)
-            self._workdir = self._workspace.new_work_dir(prefix="worker-")
-            self.local_directory = self._workdir.dir_path
-
         if not preload:
             preload = dask.config.get("distributed.worker.preload")
         if not preload_argv:
             preload_argv = dask.config.get("distributed.worker.preload-argv")
         assert preload is not None
         assert preload_argv is not None
```

Comment:
If possible, would you mind colocating this logic with the `preloading.process_preloads(...)`  call below. Generally it'd be good to keep these variable definitions close to where they're used (thinking about future code readers)

Reply:
Good call, I _think_t his ought to be okay.. there was some indirection in which order things needs to be called so maybe this got left in the wrong spot. :+1: